pandas/tests/test_generic.py

```diff
@@ -1539,6 +1540,15 @@ def test_to_xarray(self):
                            expected,
                            check_index_type=False)
 
+    def test_deepcopy_empty(self):
+        # This test covers empty frame copying with non-empty column sets
+        # as reported in issue #15370
+        # https://github.com/pandas-dev/pandas/issues/15370
+        empty_frame = DataFrame(data=[], index=[], columns=['A'])
+        empty_frame_copy = deepcopy(empty_frame)
+
+        self.assertEqual(empty_frame, empty_frame_copy)
```

Comment:
there are already generic tests in ``pandas/indexes/common.py`` but need to expand them to include using

```
from copy import copy, deepcopy

....
copy(...)
deepcopy(..)
```
these should match ``self.copy()`` and ``self.deepcopy()``